### PR TITLE
[Front] Checkboxes / selection in all "Event" categories (incidents / sightings / observed data) #2735

### PR DESCRIPTION
**Pull request by Archidoit**
Add checkboxes / selection in the 3 "Event" categories:

- incidents
- sightings
- observed data

The functionnalities linked to selection should also be added:

- selection, deselection, selectAll
- export of selected elements only
- modification of selected elements via the toolbar

### Related issues

https://github.com/OpenCTI-Platform/opencti/issues/2735

### Associated PR in connectors
https://github.com/OpenCTI-Platform/connectors/pull/1016

### Associated page in Notion
https://www.notion.so/filigran/Checkboxes-selection-in-all-Event-categories-2735-0473e635161f41d3849c8614b20d9db5#67a6b74795f5456d97d3e755fe8261eb

### CODE REVIEW

File: opencti-platform/opencti-front/src/private/components/events/Incidents.js [hunk 1]
```diff
@@ -67,6 +64,81 @@ class Incidents extends Component {
     this.setState({ openExports: !this.state.openExports });
   }
 
+  handleToggleSelectEntity(entity, event, forceRemove = []) {
```
Comment (RomuDeuxfois) [inline opencti-platform/opencti-front/src/private/components/events/Incidents.js]: Depending on the duration, this component should be migrated to use the useEntityToggle hook.
Or maybe, in the firstTime, just mutualize this code to facilitate future migration.

What do you think ?

File: opencti-platform/opencti-graphql/src/generated/graphql.ts [hunk 2]
```diff
@@ -19921,6 +19921,8 @@ export type StixDomainObjectsFiltering = {
 
 export enum StixDomainObjectsOrdering {
   Score = '_score',
+  AttributeCount = 'attribute_count',
```
Comment (RomuDeuxfois) [inline opencti-platform/opencti-graphql/src/generated/graphql.ts]: Why do we have these additions?

File: opencti-platform/opencti-front/src/private/components/events/ObservedDatas.js [hunk 3]
```diff
@@ -65,6 +64,81 @@ class ObservedDatas extends Component {
     });
   }
 
+  handleToggleSelectEntity(entity, event, forceRemove = []) {
```
Comment (RomuDeuxfois) [inline opencti-platform/opencti-front/src/private/components/events/ObservedDatas.js]: Same here

**Comment by RomuDeuxfois** [comment]
Bug found : 
- Incident entity > The select all checkbox not update the selected elements number correctly
image
- Incident entity > The unselect all checkbox still check when we unselect a single chexkbox
image
- Incident entity > The select all checkbox not take all the elements on bulk actions
image

- Sighting entity > The unselect all checkbox still check when we unselect a single chexkbox

- Observed data entity > The unselect all checkbox still check when we unselect a single chexkbox
- Observed data entity > The select all checkbox not take all the elements on bulk actions
image

File: opencti-platform/opencti-graphql/src/generated/graphql.ts (hunk 2 above)
Comment (Archidoit) [reply opencti-platform/opencti-graphql/src/generated/graphql.ts]: it is taken into account in the export if the list is sorted by those columns. The export is not possible (raise error) if the column is not in StixDomainObjectsOrdering.

File: opencti-platform/opencti-front/src/private/components/events/ObservedDatas.js (hunk 3 above)
Comment (Archidoit) [reply opencti-platform/opencti-front/src/private/components/events/ObservedDatas.js]: same

File: opencti-platform/opencti-front/src/private/components/events/Incidents.js (hunk 1 above)
Comment (Archidoit) [reply opencti-platform/opencti-front/src/private/components/events/Incidents.js]: yes good idea! what do you mean by 'mutualize' ? :)

Comment (RomuDeuxfois) [reply opencti-platform/opencti-front/src/private/components/events/Incidents.js]: I mean, put the logic in a file to use it, like the useEntityToggle file but in the old way.

Comment (Archidoit) [reply opencti-platform/opencti-front/src/private/components/events/Incidents.js]: ah yes ok I see !!

**Comment by Archidoit** [comment]
The unselect all checkbox still check when we unselect a single chexkbox --> this behavior is present everywhere (special bug to fix maybe later in another issue)

File: opencti-platform/opencti-front/src/private/components/events/Incidents.js (hunk 1 above)
Comment (Archidoit) [reply opencti-platform/opencti-front/src/private/components/events/Incidents.js]: waiting for the Incident refacto of Marie (we'll just have to add the useEntityToggle hook after the migration

Comment (RomuDeuxfois) [reply opencti-platform/opencti-front/src/private/components/events/Incidents.js]: So we waiting the merge of the Marie's PR
Good for me

**Comment by RomuDeuxfois** [comment]
Bug founds : 
- The observed data screen is broken
image

**Comment by Archidoit** [comment]
@RomuDeuxfois the error is not present for me...

**Review by RomuDeuxfois — APPROVED** [review]

**Pull request merged**